library/core/src/mem/maybe_uninit.rs

```diff
@@ -252,6 +252,41 @@ use crate::{fmt, intrinsics, ptr, slice};
 ///     std::process::exit(*code); // UB! Accessing uninitialized memory.
 /// }
 /// ```
+///
+/// # Validity
+///
+/// A `MaybeUninit<T>` has no validity requirement – any sequence of [bytes][reference-byte] of the
+/// appropriate length, initialized or uninitialized, are a valid representation of `MaybeUninit<T>`.
+///
+/// However, "round-tripping" via `MaybeUninit` does not always result in the original value.
+/// `MaybeUninit` can have padding, and the contents of that padding are not preserved.
+/// Concretely, given distinct `T` and `U` where `size_of::<T>() == size_of::<U>()`, the following
+/// code is not guaranteed to be sound:
+///
+/// ```rust,no_run
+/// # use core::mem::{MaybeUninit, transmute};
+/// # struct T; struct U;
+/// fn identity(t: T) -> T {
+///     unsafe {
+///         let u: MaybeUninit<U> = transmute(t);
+///         transmute(u)
+///     }
+/// }
+/// ```
+///
+/// If the representation of `t` contains initialized bytes at byte offsets where `U` contains padding bytes, these
+/// may not be preserved in `MaybeUninit<U>`. Interpreting the representation of `u` at type `T` again (i.e., `transmute(u)` above) may thus
```

Comment:
```suggestion
/// may not be preserved in `MaybeUninit<U>`. Interpreting the representation of `u` as type `T` again (i.e., `transmute(u)` above) may thus
```

Reply:
"interpret u as type T" sound wrong. We don't interpret u as a type, it's not a type. We use the type to tell us how to interpret u.

But I am not sure if there is widely-used standard terminology here, "interpret x at type y" just sounded most natural to me.

Reply:
Right.  Reading more closely, to use "as" here I'd instead use "as of", e.g. "interpret the bytes of `x` as being of the type `T`".

I'm pretty happy to use PLT jargon in general, but for that to work we need to use the piece of jargon pervasively enough that people pick up on it (and probably discuss the jargon in our [notation](https://doc.rust-lang.org/nightly/reference/notation.html) guide or elsewhere).  If we just use it in one place, people will just think it's a typo.

Reply:
That also works for me. I'd rather say "Transmuting `u` back to `T`" (to preserve the "again" in the current version).

Reply:
Done!